### PR TITLE
Datastore

### PR DESCRIPTION
**Pull request by j-vent**
I added all the week 3 stuff into one large PR, though I have smaller branches for tutorials 1-4. Style isn't very good at the moment, I just wanted to get the MVP out for review.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -23,10 +33,55 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
```
Comment (landondare) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:33]: You should update some of these comments to describe what the file is doing now that you have modified it.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -23,10 +33,55 @@
...
+          long timestamp = (long) entity.getProperty("timestamp");
+
+          Comment comment = new Comment(id,text,timestamp);
+          comments.add(comment);
+        }
+        else{
+            break;
+        } 
+    }
+    // response.sendRedirect("/index.html");
```
Comment (landondare) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:61]: Clean up these commented out pieces of code.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+@WebServlet("/delete-comment")
+public class DeleteCommentServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    long id = Long.parseLong(request.getParameter("id"));
+
+    Key commentEntityKey = KeyFactory.createKey("Comment", id);
+    System.out.println("Do Post");
```
Comment (landondare) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java:37]: You can use print statements like these for debugging purposes, but you should remove them once you have the logic working correctly.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -37,3 +37,50 @@ window.addEventListener("scroll", function() {
   }
 });
 
+async function getComments() {
+  console.log("in get comments");
```
Comment (landondare) [inline portfolio/src/main/webapp/script.js:41]: Delete these log statements once you have the functionality working properly.

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -37,3 +37,50 @@ window.addEventListener("scroll", function() {
...
+  const numComments = document.getElementById("numComments").value;
+  const response = await fetch('/data?numComments='+ numComments);
```
Comment (landondare) [inline portfolio/src/main/webapp/script.js:43]: What you are doing here with the await function is blocking the current thread until the fetch function finishes. It's removing the async logic from this code. You should instead provide a "callback function" that should execute once this operation completes. You set a callback with the `.then()` function. You can reorganize your code into something like
```
fetch('/data?numComments='+ numComments)
  .then(response => response.json())
  .then((comments) => {
    const commentSection = document.getElementById('comment-list');
    commentSection.innerText = ""; // clear old comments 
    comments.forEach((comment) => {
      commentSection.appendChild(createCommentElement(comment));
    }
);

```

Also check out this page for another example of using promises
https://github.com/googleinterns/step/blob/master/walkthroughs/week-3-server/examples/subtraction-game/src/main/webapp/script.js#L19

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -37,3 +37,50 @@ window.addEventListener("scroll", function() {
...
+    const commentElement = document.createElement("li");
+    const textElement = document.createElement("span");
+    textElement.innerText = comment.text;
+    commentElement.appendChild(textElement);
+    return commentElement;
+}
+
+async function deleteComments(){
+  console.log("in delete comments");
+  const response = await fetch('/data?numComments=10');
```
Comment (landondare) [inline portfolio/src/main/webapp/script.js:67]: Replace these await comments with async logic using the `.then()` function.

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -37,3 +37,50 @@ window.addEventListener("scroll", function() {
...
+ 
+  console.log(comments);
+  console.log(numComments);
+  
+  comments.forEach((comment) => {
+      commentSection.appendChild(createCommentElement(comment));
+  });
+}
+
+function createCommentElement(comment){
```
Comment (landondare) [inline portfolio/src/main/webapp/script.js:57]: See if you can figure out how to add a delete button to each comment. You can probably add a button to that will call your "/delete-comment" action with the id of the current comment list item. Try playing around in the chrome debug tools and see what it takes to add a button to a list item. Or you could even add new `<div>` tags instead!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (j-vent) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:33]: My bad, I didn't clean up comments. Same for the other four comments below.

**Pull request merged**